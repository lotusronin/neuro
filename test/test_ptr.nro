import stdio;

fn main() : u32 {
    num : u32 = 0;
    stdio::puts("num is");
    stdio::printi(num);
    stdio::putchar('\n');
    add1(&num);
    stdio::puts("num is now");
    stdio::printi(num);
    stdio::putchar('\n');
    return 0;
}

fn add1(x : *u32) : void {
    @x = @x + 1;
    return;
}
