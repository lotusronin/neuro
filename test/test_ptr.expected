num is
0
num is now
1
